actions/helm-lint/action.yaml

```diff
@@ -1,31 +1,45 @@
 name: "Lint helm chart"
-description: "Lint all helm charts in the chart directory (default: charts)"
+description: "Lint all helm charts in the charts directory of a repository"
 inputs:
-  lintConfig:
-    description: "The path to the lint configuration file."
+  ref:
+    description: "The ref name to checkout the repository."
+    required: false
+    default: ${{ github.event.repository.default_branch }}
+    type: string
+  lint-config-path:
+    description: "The path to the lint configuration file (For an example see https://github.com/helm/chart-testing/blob/main/pkg/config/test_config.yaml)."
+    required: false
     default: ".github/lint-config.yaml"
+    type: string
+  helm-version:
+    description: "The helm version for setting up poetry."
     required: false
+    default: "v3.4.0"
+    type: string
+
 runs:
   using: "composite"
   steps:
-    - name: Set up Helm
-      uses: azure/setup-helm@v1
+    - name: Check out repository
+      uses: actions/checkout@v2
       with:
-        version: "v3.4.0"
+        ref: "${{ inputs.ref }}"
+        # fetch-depth of 2 needed for comparison of helm charts between commits while linting
+        fetch-depth: 2
 
-    - name: Set up python
-      uses: actions/setup-python@v2
+    - name: Set up Helm
+      uses: azure/setup-helm@v1
       with:
-        python-version: "3.8"
+        version: "${{ inputs.helm-version }}"
 
     - name: Set up chart-testing
       uses: helm/chart-testing-action@v2.0.1
 
     - name: Run chart-testing (lint)
       run: |
         if [[ "${GITHUB_REF#refs/heads/}" == "main" || "${GITHUB_REF#refs/heads/}" == "master" ]]; then
-          ct lint --config ${{ inputs.lintConfig }} --since HEAD^
+          ct lint --config ${{ inputs.lint-config-path }} --since HEAD^
         else
-          ct lint --config ${{ inputs.lintConfig }} --since HEAD
+          ct lint --config ${{ inputs.lint-config-path }} --since HEAD
```

Comment:
Why since head? Don't we want to compare to main/master on a feature branch?

Reply:
Yes you are right, I don't know what my thought process was when I merged this in the past. But it should be pointing to the head of the default branch

Reply:
As far as I can tell, this is the default for `--since`

Reply:
In the following link it says, that the default is be `HEAD`:
https://github.com/helm/chart-testing/blob/main/doc/ct_lint.md

Reply:
Actually, I think your thought process was absolutely correct. Looking at the docs, the target-branch is master (and I guess they also support main). It should therefore work for a feature-branch, because it compares to the HEAD of main, which is exactly what we want.